### ISSUE TITLE
$spread signature should require an object as anything else is nonsensical

### ISSUE THREAD

**Issue by wnm3** [jsonata-js/jsonata#313]
The signature for spread is <x-:a<o>> which allows for odd returns like:
$spread("a") -> "a"
$spread(1) -> 1
$spread([]) -> null
$spread([1]) -> [1]
I think the initial parameter should be o for object and reject any of these cases as a bad arg1...
e.g., <o-:a<o>>

**Comment by wnm3** [comment]
In particular, I would remove the else { return = arg; } from below and change the signature to reject anything but an object to throw the T0410 exception.

    /**
     * Splits an object into an array of object with one property each
     * @param {*} arg - the object to split
     * @returns {*} - the array
     */
    function spread(arg) {
        var result = createSequence();

        if (Array.isArray(arg)) {
            // spread all of the items in the array
            arg.forEach(function (item) {
                result = append(result, spread(item));
            });
        } else if (arg !== null && typeof arg === 'object' && !isLambda(arg)) {
            for (var key in arg) {
                var obj = {};
                obj[key] = arg[key];
                result.push(obj);
            }
        // } else {
            // result = arg;
        }
        return result;
    }

**Comment by andrew-coleman** [comment]
Yes, I think I agree with this - certainly the last bit about removing the final else clause.
